lib/modules/platform/gitlab/index.ts

```diff
@@ -1005,21 +1008,29 @@ export async function setBranchStatus({
     options.target_url = targetUrl;
   }
 
-  if (branchSha) {
+  const retryTimes = parseInteger(
+    process.env.RENOVATE_X_GITLAB_BRANCH_STATUS_CHECK_ATTEMPTS,
+    2,
+  );
+
+  for (let attempt = 1; attempt <= retryTimes; attempt += 1) {
```

Comment:
the whole loop should have a try catch and early return on failures from gitlab API